mutyper/cli.py

```diff
@@ -61,6 +59,19 @@ def copy_fasta(file, outfile):
         copyfile(file, outfile)
 
 
+def iterate_with_ambiguity_warning(vcf):
+    """In several places we want to check for genotype ambiguity as we iterate
+    over vcf variants, so we define this wrapper generator."""
+    for variant in vcf:
+        yield variant
+        if variant.num_unknown:
+            logging.warning(
+                "Ambiguous genotypes found! Continuing by assuming reference genotypes for these variants."
+            )
+            break
+    yield from vcf
+
+
```

Comment:
Looking at the code in cyvcf2 it looks like for this function to work strict must be set to true which you do so that is great. This code also checks for `==2` and may not work with `gts012=True`, which you do not have set so it is also okay. I guess I am just worried that this function will fail if the vcf is opened with these params set incorrectly. Should we add a warning to the doc string about how cyvcf2 must be opened i.e `vcf = cyvcf2.VCF(args.vcf, strict_gt=True)`.

Or maybe the function should do the file opening itself? IDK.  
```
property num_unknown:
        "number unknown samples at this variant."
        def __get__(self):
            if self._gt_types == NULL:
                self.gt_types
            cdef int n = 0, i = 0
            for i in range(self.vcf.n_samples):
                if self._gt_types[i] == 2:
                    n+=1
            return n
```

Reply:
Yes I agree we should check for this. The cyvcf2 code seems to indicate that the VCF object should have the attributes we need to check: https://github.com/brentp/cyvcf2/blob/15b7de47914418019ab472417a87562f69ca7206/cyvcf2/cyvcf2.pyx#L266-L268
However, I get attribute errors when I try to access these.

Also, isn't this behavior a cyvcf2 bug? I would expect `num_unknown` to know about the `gts012` parameter, so I would expect the following two loops to output the same thing:
```python
>>> import cyvcf2
>>> vcf = cyvcf2.VCF("tests/test_data/snps.missing_gts.vcf", strict_gt=True, gts012=False)
>>> for variant in vcf:
...     print(variant.num_unknown)
... 
0
0
1
1
1
>>> vcf = cyvcf2.VCF("tests/test_data/snps.missing_gts.vcf", strict_gt=True, gts012=True)
>>> for variant in vcf:
...     print(variant.num_unknown)
... 
0
0
0
0
0
```

Reply:
I agree this works for our purposes. I posted a cyvcf2 issue: https://github.com/brentp/cyvcf2/issues/236

Reply:
Great, can you make the update to the PR? or do you need me to?

Reply:
I moved this inside the spectra function, and removed the argument so it's harder to use it incorrectly, and added a comment about the cyvcf2 bug.